Foundation/include/Poco/PIDFile.h

```diff
@@ -0,0 +1,101 @@
+//
+// PIDFile.h
+//
+// Library: Foundation
+// Package: Processes
+// Module:  PIDFile
+//
+// Definition of the PIDFile class.
+//
+// Copyright (c) 2023, Applied Informatics Software Engineering GmbH.
+// Aleph ONE Software Engineering d.o.o.,
+// and Contributors.
+//
+// SPDX-License-Identifier:    BSL-1.0
+//
+
+
+#ifndef Foundation_PIDFile_INCLUDED
+#define Foundation_PIDFile_INCLUDED
+
+
+#include "Poco/Foundation.h"
+#include <memory>
+
+
+namespace Poco {
+
+
+class Foundation_API PIDFile
+	/// A utility class, creating process ID file on
+	/// construction and deleting it on destruction.
+{
+public:
+	using Ptr = std::unique_ptr<PIDFile>;
+
+	static const int INVALID_PID = -1;
+
+	PIDFile();
+		/// Creates the PIDFile.
+
+	PIDFile(const std::string& fileName, bool write = true);
+		/// Creates the PIDFile.
+		/// If `fileName` is not empty, creates the PID file.
+		/// If `write` is true, the file is written.
+
+	~PIDFile();
+		/// Destroys the PIDFile.
+		/// If fileName is not empty, deletes the PID file.
+
+	const std::string& getName() const;
+		/// Returns the file name.
+
+	void setName(const std::string& fileName);
+		/// Sets the file name;
```

Comment:
comments should not end with `;` - besides being grammatically incorrect, it causes `CodeQL` to create alerts

Reply:
Fixed in [1b1feaf](https://github.com/pocoproject/poco/pull/4225/commits/1b1feaf5b4abf5f765e5a9cc656e70deb9b0813c)